libcontainer/cgroups/fs/apply_raw.go

```diff
@@ -270,6 +270,9 @@ func (raw *cgroupData) path(subsystem string) (string, error) {
 	}
 
 	cgPath := filepath.Join(raw.parent, raw.name)
+	if raw.config.CgroupsPath != "" {
```

Comment:
It won't break compatibility, Docker can still use parent and name, but before it call runc apis, Docker needs to combine parent and name to CgroupPath. And yes, I think parent and name also have the restriction to be relative to cgroups mount point.